### ISSUE TITLE
Override tmp folder

### ISSUE THREAD

**Issue by vkostyukov** [jvm-profiling-tools/async-profiler#147]
When running JVM in a containerized environment, it gets an assigned temp folder that is not `/tmp`. This doesn't allow async-profiler to attach to a process with an error:

```
Could not start attach mechanism: No such file or directory
```

I wonder if there is a possibility override a `/tmp` folder location when running an async-profiler so we can point it to over in our isolated file system.

**Comment by apangin** [comment]
Make sure you use the latest async-profiler/jattach, and run the profiler under root.
When run under privileged user, jattach automatically switches to the proper mount namespace.
It is known to work with docker containers.

`build/jattach --version` should say at least `1.3`

**Comment by vkostyukov** [comment]
Thanks @apangin! Yea, I'm pretty sure we're on 1.3. I will try sudo/root and let you know.

**Comment by vkostyukov** [comment]
Sorry forgot to mention: it's not Docker though. It's Aurora/Mesos. I'm not sure how different it is from Docket as far as the container management goes, but I will give the sudo/rot a try.

**Comment by apangin** [comment]
I realized that different location of `/tmp` does not necessarily imply different mount namespaces. If a process runs under `chroot`, it sees different directory structure, yet it works in the same namespace.

Fortunately, there is no need to override the directory manually. We can just look into `/proc/pid/root/tmp` instead of `/tmp`. This works even for unprivileged user, if profiler is launched by the owner of the target process.

I've committed the fix. Please check if it helps.

**Comment by vkostyukov** [comment]
To close the loop on this. I built async-profiler from sources and it seems to be attaching successfully now (even when being run under unprivileged user)! Thanks a ton, @apangin!

**Comment by apangin** [comment]
Thanks for confirming it works.

**Comment by apangin** [comment]
I reworked the fix due to #158. I verified that profiler is still able to connect to chroot'ed processes, but I kindly invite you to re-check that it works in your case as well. Thanks.

**Comment by vkostyukov** [comment]
> I reworked the fix due to #158. I verified that profiler is still able to connect to chroot'ed processes, but I kindly invite you to re-check that it works in your case as well. Thanks.

Hey @apangin! I compiled the new version of async-profiler from master and verified it still works on our environment. Everything seems fine. Thanks!
